gammapy/modeling/covariance.py

```diff
@@ -210,3 +210,32 @@ def __str__(self):
 
     def __array__(self):
         return self.data
+
+
+class CovarianceMixin:
+    """Mixin class for covariance property on multi-components models"""
+
+    def _check_covariance(self):
+        if not self.parameters == self._covariance.parameters:
+            self._covariance = Covariance.from_stack(
+                [model.covariance for model in self._models]
+            )
+
+    @property
+    def covariance(self):
+        """Covariance as a `~gammapy.modeling.Covariance` object."""
+        self._check_covariance()
+
+        for model in self._models:
+            self._covariance.set_subcovariance(model.covariance)
+
+        return self._covariance
+
+    @covariance.setter
+    def covariance(self, covariance):
```

Comment:
Can you add a docstring defining the type of `covariance`

Reply:
The setter just display the same output as its associated property and the type of `Covariance` is described there. I think we have the same behaviour everywhere. Probably better not change too much docstring in a bugfix that has to beck backported.